### PR TITLE
M1 support on MacStadium

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes RNBW-####

## What changed (plus any additional context for devs)
In order to run the latest xcode + iOS  with the new bitrise M1 setup I had to do  a couple of things:

- Bump detox to latest version which includes a fix for iOS 15.5
- Disable keychain options for testing to work around [this issue](https://github.com/oblador/react-native-keychain/issues/509) 
- - Minor refactor for e2e

There are reports that [this has been fixed on iOS 16 beta 1](https://developer.apple.com/forums/thread/685773?page=2) so we should be able to get rid of it soon.

## PoW (screenshots / screen recordings)
CI should be passing on the m1 stack

## Dev checklist for QA: what to test
Make sure the biometrics works as expected when IS_TESTING=false 

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: scripts/setup-ci.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+# fail if any commands fails
+set -e
+# debug log
+set -x
+
+# Yarn sucks and running a couple of times does the trick.
+# See https://github.com/yarnpkg/yarn/issues/7212 before yelling at me
```
Comment (skylarbarrera) [inline scripts/setup-ci.sh:8]: 😂

**Review by skylarbarrera — APPROVED** [review]
🔥

**Review by ibrahimtaveras00 — APPROVED** [review]
Double checked with IS_TESTING=false for backing up and viewing secret phrase & private key as well as sending txn, and confirming via WC on both OS's

QA Passed 👍🏽

**Review by nickbytes — APPROVED** [review]

**Pull request merged**